node/backend-read_products/pages/api/products/[id].js

```diff
@@ -1,13 +1,17 @@
-import { products } from "../../../lib/products";
+import dbConnect from "@/db/connect";
+import Product from "@/db/models/Product";
 
-export default function handler(request, response) {
+export default async function handler(request, response) {
+  await dbConnect();
   const { id } = request.query;
 
-  const product = products.find((product) => product.id === id);
+  if (request.method === "GET") {
+    const product = await Product.findById(id);
```

Comment:
the task says that we should use `populate` here, so i did `const product = await Product.findById(id).populate("reviews");`